c++/triqs_cthyb/solver_core.cpp

```diff
@@ -154,6 +154,16 @@ namespace triqs_cthyb {
     for (int b : range(gf_struct.size()))
       max_imag = std::max(max_imag, max_element(abs(imag(Delta_infty_vec[b]))));
 
+    // set off diagonal terms in Delta_infty to 0 if they are below off_diag_threshold
+    for (int b : range(gf_struct.size())){
+      for (int i : range(first_dim(Delta_infty_vec[b]))){
+        for (int j : range(second_dim(Delta_infty_vec[b]))){
+          if (i != j && abs(Delta_infty_vec[b](i,j)) < params.off_diag_threshold)
+            Delta_infty_vec[b](i,j) = 0;
+        }
+      }
+    }
+
```

Comment:
To be consistent with the `imag_threshold` behavior I suggest to only set the corresponding terms to zero in the h_loc operator, not in `Delta_infty_vec`
```suggestion
```

For this it should be sufficient to add an additional check in the h_loc initialization below line 182

```cpp
  if(n1 != n2 && abs(Delta_infty_vec[b](n1, n2)) > params.off_diag_threshold.value_or(0.0)) e_ij = 0.0;
```